service/src/test/java/com/kodemon/service/test/facade/FightFacadeTest.java

```diff
@@ -0,0 +1,301 @@
+        pokemonService = mock(PokemonService.class);
+        badgeService = mock(BadgeService.class);
+        timeService = mock(TimeService.class);
+        fightFacade = new FightFacadeImpl(beanMappingService, trainerFightService, pokemonFightService, pokemonService,
+                trainerService, badgeService, timeService);
+
+        prepareTrainers();
+        prepareGym();
+        prepareFights();
+    }
+
+    @Test
+    public void fightForBadgeTest() {
+        Date today = new Calendar.Builder().setDate(2015, 4, 1).build().getTime();
+
+        when(trainerFightService.wasFightForBadgeSuccessful(challenger, defender)).thenReturn(true);
+        Badge badge = new Badge();
+        badge.setGym(targetGym);
+        when(badgeService.createBadgeOfGym(targetGym)).thenReturn(badge);
+        when(timeService.currentDate()).thenReturn(today);
+        fightFacade.fightForBadge(challengerDTO, targetGymDTO);
+
+        FightDTO newFightDTO = new FightDTO();
+        newFightDTO.setTargetGym(targetGymDTO);
+        newFightDTO.setChallenger(challengerDTO);
+        newFightDTO.setFightTime(today);
+        newFightDTO.setWasChallengerSuccessful(true);
+
+        TrainerFight newFight = new TrainerFight();
+        newFight.setTargetGym(targetGym);
+        newFight.setChallenger(challenger);
+        newFight.setFightTime(today);
+        newFight.setWasChallengerSuccessful(true);
+
+        when(beanMappingService.mapTo(newFightDTO, TrainerFight.class)).thenReturn(newFight);
+
+        Mockito.verify(badgeService, times(1)).assignTrainerToBadge(challenger, badge);
+        Mockito.verify(trainerService, times(1)).addBadge(badge, challenger);
+    }
+
+    @Test
+    public void fightWildPokemonTest() {
+        Pokemon randomWildPokemon = new Pokemon();
+        randomWildPokemon.setName(PokemonName.ABRA);
+        randomWildPokemon.setLevel(1);
+        randomWildPokemon.setNickname("Teleportabra");
+        when(pokemonService.generateWildPokemon(null)).thenReturn(randomWildPokemon);
+
+        List<Pokemon> trainersPokemons = new ArrayList<>();
+        trainersPokemons.add(pikachu);
+        when(pokemonService.findByTrainer(challenger)).thenReturn(trainersPokemons);
+        when(pokemonFightService.getScorePair(pikachu, randomWildPokemon)).thenReturn(new Pair<Double, Double>(20.0, 1.0));
+
+        fightFacade.fightWildPokemon(challengerDTO, WildPokemonFightMode.TRAIN);
+        Mockito.verify(pokemonService, times(1)).levelPokemonUp(pikachu);
+    }
+
+    @Test
+    public void listFightsBetweenTest() {
+        Date from = new Calendar.Builder().setDate(2015, 4, 1).build().getTime();
+        Date to = new Calendar.Builder().setDate(2015, 4, 3).build().getTime();
+        when(trainerFightService.findByFightTimeBetween(from, to)).thenReturn(trainerFights);
+        List<FightDTO> fights = fightFacade.listFightsBetween(from, to);
+        assertThat(fights.size(), is(2));
```

Comment:
Here, you should check whether you received list with instances you wanted, but it should be simple test, so I think that this is the definition of simplicity :smile: and nothing more is needed